gnome-initial-setup/pages/account/um-photo-dialog.c

```diff
@@ -195,36 +192,58 @@ create_face_widget (gpointer item,
 }
 
 static GSList *
-get_facesdirs (void)
+get_settings_facesdirs (void)
 {
+        g_autoptr(GSettings) settings = NULL;
+        g_autoptr(GSettingsSchema) schema = NULL;
+        g_auto(GStrv) keys = NULL;
+
         GSList *facesdirs = NULL;
-        const gchar * const * data_dirs;
+        char **settings_dirs;
         int i;
-        const char *facesdir_env;
-        GisDriver *driver;
+        gboolean key_found = FALSE;
+
+        GSettingsSchemaSource *source = g_settings_schema_source_get_default ();
+
+        if (!source)
+                return facesdirs;
+
+        schema = g_settings_schema_source_lookup (source,
+                                                  "org.gnome.ControlCenter",
+                                                  FALSE);
 
-        facesdir_env = g_getenv ("GIS_ACCOUNT_PAGE_FACESDIR");
-        if (facesdir_env != NULL && g_strcmp0 (facesdir_env, "") != 0) {
-                facesdirs = g_slist_prepend (facesdirs,
-                                             g_strdup (facesdir_env));
+        keys = g_settings_schema_list_keys (schema);
```

Comment:
[`g_settings_schema_has_key()`](https://developer.gnome.org/gio/stable/gio-GSettingsSchema-GSettingsSchemaSource.html#g-settings-schema-has-key)?

Reply:
Yes, sure that's better :sweat_smile: I was looking for that in the doc, but I didn't find it because I was looking here: https://developer.gnome.org/gio/stable/GSettings.html